File: C++/level2/가장_큰_정사각형_찾기.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<vector<int>> board) {
    int answer  = 0;

    if(board.size() == 1 || board[0].size() == 1){
        for(int i=0; i<board.size(); i++)
            for(int j=0; j<board[0].size(); j++)
                if(board[i][j] == 1)
                    return 1;
        return 0;
    }

    for(int i=1; i<board.size(); i++){
        for(int j=1; j<board[0].size(); j++){
            if(board[i][j] == 1){
                board[i][j] = min({ board[i][j-1], board[i-1][j], board[i-1][j-1]}) + 1;
                answer = max(answer, board[i][j]);
            }
        }
    }

    return answer*answer;
}

File: JavaScript/level2/가장_큰_정사각형_찾기.js
function solution(board)
{
    var answer = 0;
  
    if(board[0][0] === 1)
        answer++;
    
    for(let i=1; i<board.length; i++)
        for(let j=1; j<board[0].length; j++)
            if(board[i][j] === 1){
                board[i][j] = Math.min(board[i-1][j-1], board[i-1][j], board[i][j-1]) + 1;
                answer = Math.max(answer, board[i][j]);
            }

    return Math.pow(answer, 2);
}